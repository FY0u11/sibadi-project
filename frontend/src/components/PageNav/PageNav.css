.my-page-nav {
  margin: 0 auto 20px auto;
  width: 200px;
}
