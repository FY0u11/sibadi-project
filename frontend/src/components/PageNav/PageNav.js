import React from 'react'

import './PageNav.css'
import Button from '../Button'

const PageNav = ({ page, setPage, isNextDisabled }) => {
  return (
    <div className="my-page-nav">
      <Button disabled={page === 1} action={() => setPage(page - 1)}>
        &lt;
      </Button>
      <Button disabled>{page}</Button>
      <Button disabled={isNextDisabled} action={() => setPage(page + 1)}>
        &gt;
      </Button>
    </div>
  )
}

export default PageNav
