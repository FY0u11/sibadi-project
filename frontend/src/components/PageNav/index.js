import React from 'react'
import { connect } from 'react-redux'

import { setPage } from '../../store/components/pageNav/actions'

import PageNav from './PageNav'

const PageNavContainer = ({ page, settings, amountOfProducts, setPage }) => (
  <PageNav
    page={page}
    setPage={setPage}
    isNextDisabled={page === Math.ceil(amountOfProducts / settings.pageLen)}
  />
)

const putStateToProps = state => ({
  page: state.pageNav.page,
  settings: state.root.settings
})

const putActionsToProps = { setPage }

export default connect(putStateToProps, putActionsToProps)(PageNavContainer)
