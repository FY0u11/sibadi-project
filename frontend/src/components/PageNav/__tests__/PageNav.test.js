import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import PageNav from '../PageNav'

afterEach(cleanup)

describe('Testing Card component', () => {
  it('It should be properly rendered with prev button in disabled state', () => {
    const { asFragment, getByText } = render(<PageNav page={1} />)
    expect(asFragment()).toMatchSnapshot('1')
    expect(getByText('<')).toBeDisabled()
    expect(getByText('>')).toBeEnabled()
  })
  it('It should be properly rendered with next button in disabled state', () => {
    const { asFragment, getByText } = render(<PageNav page={2} isNextDisabled={true} />)
    expect(asFragment()).toMatchSnapshot('2')
    expect(getByText('>')).toBeDisabled()
    expect(getByText('<')).toBeEnabled()
  })
  it('It should be properly rendered with next and prev buttons both in disabled state', () => {
    const { asFragment, getByText } = render(<PageNav page={1} isNextDisabled={true} />)
    expect(asFragment()).toMatchSnapshot('3')
    expect(getByText('>')).toBeDisabled()
    expect(getByText('<')).toBeDisabled()
  })
  it('It should be properly rendered with next and prev buttons both in enabled state', () => {
    const { asFragment, getByText } = render(<PageNav page={2} />)
    expect(asFragment()).toMatchSnapshot('4')
    expect(getByText('>')).toBeEnabled()
    expect(getByText('<')).toBeEnabled()
  })
  it('Prev and next buttons should decrement and increment the current page respectively', () => {
    let page = 1
    const callback = jest.fn(v => {
      page = v
    })
    const { getByText, rerender } = render(<PageNav page={page} setPage={callback} />)
    expect(page).toBe(1)
    const nextButton = getByText('>')
    fireEvent.click(nextButton)
    expect(page).toBe(2)
    rerender(<PageNav page={page} setPage={callback} />)
    expect(page).toBe(2)
    const prevButton = getByText('<')
    fireEvent.click(prevButton)
    expect(page).toBe(1)
  })
})
