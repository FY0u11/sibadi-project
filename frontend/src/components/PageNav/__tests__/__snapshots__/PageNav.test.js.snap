// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component It should be properly rendered with next and prev buttons both in disabled state: 3 1`] = `
<DocumentFragment>
  <div
    class="my-page-nav"
  >
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      &lt;
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      1
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;

exports[`Testing Card component It should be properly rendered with next and prev buttons both in enabled state: 4 1`] = `
<DocumentFragment>
  <div
    class="my-page-nav"
  >
    <button
      class="btn btn-dark"
      style="margin-right: 20px;"
    >
      &lt;
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      2
    </button>
    <button
      class="btn btn-dark"
      style="margin-right: 20px;"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;

exports[`Testing Card component It should be properly rendered with next button in disabled state: 2 1`] = `
<DocumentFragment>
  <div
    class="my-page-nav"
  >
    <button
      class="btn btn-dark"
      style="margin-right: 20px;"
    >
      &lt;
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      2
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;

exports[`Testing Card component It should be properly rendered with prev button in disabled state: 1 1`] = `
<DocumentFragment>
  <div
    class="my-page-nav"
  >
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      &lt;
    </button>
    <button
      class="btn btn-dark"
      disabled=""
      style="margin-right: 20px;"
    >
      1
    </button>
    <button
      class="btn btn-dark"
      style="margin-right: 20px;"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;
